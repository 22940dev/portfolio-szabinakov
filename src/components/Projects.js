import React from "react";
import "../styles/Projects.css";

import Checkout from "./CheckOut";

import cruise from "../pictures/cruise.png";
import bands from "../pictures/BandS.png";
import pet from "../pictures/pet.png";
import weather from "../pictures/weather.png";
import nasa from "../pictures/nasa.png";
import surreal from "../pictures/SurrealE.png";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faAnchor,
  faCloudSun,
  faDog,
  faRocket,
  faUsers,
} from "@fortawesome/free-solid-svg-icons";
import { faFortAwesome } from "@fortawesome/free-brands-svg-icons";
const Projects = () => {
  return (
    <>
      <Checkout />
      <div data-testid="projectContainer" className="projectContainer">
        <div className="card">
          <div className="cardInner">
            <div data-testid="bringAndShareFront" className="cardFront">
              <h3>Bring and Share</h3>
              <FontAwesomeIcon className="iconContainer" icon={faUsers} />
            </div>

            <div data-testid="bringAndShareBack" className="cardBack">
              <img src={bands} alt="Bring and Share" />
              <p className="projectPhar">
                This app has been a group project where I have been fully
                responsible to set up the API, database, and write tests. The
                App is allow you to create and event and add on participants
                with their diatry requriments. I have created the component for
                adding participants, and the participant list with all their
                details.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/Bring-and-Share"
                  target="blank"
                >
                  Front-End
                </a>
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/bring-and-share-api"
                  target="blank"
                >
                  Back-End
                </a>
                <a
                  className="aCardLink"
                  href="https://bring-and-share-final-project.vercel.app/"
                  target="blank"
                >
                  Website
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="card">
          <div className="cardInner">
            <div data-testid="surrealFront" className="cardFront">
              <h3>Surreal Estate</h3>
              <FontAwesomeIcon className="iconContainer" icon={faFortAwesome} />
            </div>
            <div data-testid="surrealBack" className="cardBack">
              <img src={surreal} alt="Surreal Estate" />
              <p className="projectPhar">
                It is a property search engine just like Zoopla or Rightmove.
                There are properties, favourites and add property pages.The user
                is able to log in with their Facebook account. Once logged in,
                you are able to save properties to your favourites or remove
                them.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/surreal-estate"
                  target="blank"
                >
                  View Source
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="card">
          <div className="cardInner">
            <div data-testid="weatherFront" className="cardFront">
              <h3>Weather App</h3>
              <FontAwesomeIcon className="iconContainer" icon={faCloudSun} />
            </div>
            <div data-testid="weatherBack" className="cardBack">
              <img src={weather} alt="Weather App Project" />
              <p className="projectPhar">
                By using external data the app allows the user to search by
                cities. Once you type in the requested city it displays 5 days
                of weather forecast. By clicking on More Details, it gives you a
                more detailed description of the weather of the chosen day.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/weather-app"
                  target="blank"
                >
                  View Source
                </a>
                <a
                  className="aCardLink"
                  href="https://weather-app-ochre-two.vercel.app/"
                  target="blank"
                >
                  Website
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="card">
          <div className="cardInner">
            <div data-testid="nasaFront" className="cardFront">
              <h3>Nasa's Picture Search</h3>
              <FontAwesomeIcon className="iconContainer" icon={faRocket} />
            </div>
            <div data-testid="nasaBack" className="cardBack">
              <img src={nasa} alt="Nasa Search Project" />
              <p className="projectPhar">
                This is my first React web application I have built. The idea is
                to search images based on a query realiting to space using the
                NASA API.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/Nasa-picture-searchengine"
                  target="blank"
                >
                  View Source
                </a>
                <a
                  className="aCardLink"
                  href="https://tech-test.vercel.app/"
                  target="blank"
                >
                  Website
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="card">
          <div className="cardInner">
            <div data-testid="cruiseFront" className="cardFront">
              <h3>Cruise Ship</h3>
              <FontAwesomeIcon className="iconContainer" icon={faAnchor} />
            </div>
            <div data-testid="cruiseBack" className="cardBack">
              <img src={cruise} alt="Cruise Ship Project" />
              <p className="projectPhar">
                At this project I have built a cruise ship which is able to
                sail. There is a board also included which tells you the current
                and the next port. You are also able to add ports. It has been
                designed to look really oldschool. Because that is what I love
                the most.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/cruise-ships"
                  target="blank"
                >
                  View Source
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="card">
          <div className="cardInner">
            <div data-testid="petFront" className="cardFront">
              <h3>Virtual Pet</h3>
              <FontAwesomeIcon className="iconContainer" icon={faDog} />
            </div>
            <div data-testid="petBack" className="cardBack">
              <img src={pet} alt="Virtual Pet Project" />
              <p className="projectPhar">
                I have built a Virtual Pet with JavaScript. Same principles like
                the good old game Tamagotchi. The Pet gets older and as it gets
                older, it gets hungrier and more and more unfit. You are able to
                take it for a walk to get back its fitness and of course feed
                it. However, nature is cruel and even if you take care of it on
                the best way the pet will D-I-E when it reaches 30 years.
              </p>
              <div className="linksCard">
                <a
                  className="aCardLink"
                  href="https://github.com/szabinakov/Virtual-Pet"
                  target="blank"
                >
                  View Source
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default Projects;
