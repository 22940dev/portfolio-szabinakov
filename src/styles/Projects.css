.projectContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    margin-top: 120px;
}

.card {
    margin-top: 40px;
    margin-bottom: 40px;
    height: 300px;
    width: 300px;
}

img {
    max-width: 200px;
    height: auto;
    border: groove 1px;
    border-radius: 1%;
}

.imgContainer {
    width: 240px;
    height: 150px;
}
.cardInner {
    position: relative;
    width: 100%;
    height: 100%;
    text-align: center;
    transition: transform 1s;
    transform-style: preserve-3d;
}

.card:hover .cardInner {
    transform: rotateY(180deg);
}

.cardFront, .cardBack {
    position: absolute;
    width: 100%;
    height: 100%;
    -webkit-backface-visibility: hidden;
    backface-visibility: hidden;
    border-radius: 3px;
    box-shadow: 0px 0px 10px;

}

.cardFront {
    background-color: #f5e6d8be;
    font-size: 18px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

.cardBack {
    background-color: #EEE4E1;
    transform: rotateY(180deg);
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: justify;
    font-size: 13px;
}

.projectPhar {
    padding-right: 20px;
    padding-left: 20px;
    height: 175px;
}
.linksCard {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
}

.aCardLink {
    text-decoration: none;
    text-align: center;
    background-color: #ddd6d4;
    color: black;
    border: solid 1px rgb(119, 119, 119);
    border-radius: 3px;
    box-shadow: 1px 1px 1px;
    width: 90px;
    height: 20px;
    padding-top: 4px;
    padding-bottom: 4px;
}

.aCardLink:hover {
    background-color: #c9bcb8;
}