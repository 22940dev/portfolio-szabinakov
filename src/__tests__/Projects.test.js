import React from 'react'
import {render} from '@testing-library/react'
import Projects from '../components/Projects'

describe('Projects', () => {
    it('matches snapshot', () => {
        const {asFragment} = render (<Projects/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders all elements correctly', () => {
        const {getByTestId} = render (<Projects/>)

        expect(getByTestId('projectContainer')).toHaveClass('projectContainer')
        expect(getByTestId('bringAndShareFront')).toHaveClass('cardFront')
        expect(getByTestId('bringAndShareBack')).toHaveClass('cardBack')
        expect(getByTestId('surrealFront')).toHaveClass('cardFront')
        expect(getByTestId('surrealBack')).toHaveClass('cardBack')
        expect(getByTestId('weatherFront')).toHaveClass('cardFront')
        expect(getByTestId('weatherBack')).toHaveClass('cardBack')
        expect(getByTestId('nasaFront')).toHaveClass('cardFront')
        expect(getByTestId('nasaBack')).toHaveClass('cardBack')
        expect(getByTestId('cruiseFront')).toHaveClass('cardFront')
        expect(getByTestId('cruiseBack')).toHaveClass('cardBack')
        expect(getByTestId('petFront')).toHaveClass('cardFront')
        expect(getByTestId('petBack')).toHaveClass('cardBack')
    })
})