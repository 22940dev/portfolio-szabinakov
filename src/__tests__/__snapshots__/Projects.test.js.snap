// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkoutContainer"
  >
    <div
      class="checkOutIntro"
    >
      <h3>
        Check out all my project by flipping the cards below!
      </h3>
      <h4>
        Tech Stack:
      </h4>
    </div>
    <div
      class="logoContainer"
      data-testid="logoContainer"
    >
      <a
        href="https://html.com/"
        target="blank"
      >
        <img
          alt="HTML-logo"
          class="logoImg"
          src="HTML-logo.png"
        />
      </a>
      <a
        href="https://www.w3.org/Style/CSS/Overview.en.html"
        target="blank"
      >
        <img
          alt="CSS-logo"
          class="logoImg"
          src="CSS-logo.png"
        />
      </a>
      <a
        href="https://www.javascript.com/"
        target="blank"
      >
        <img
          alt="Javascript-logo"
          class="logoImg"
          src="JS-logo.png"
        />
      </a>
      <a
        href="https://reactjs.org/"
        target="blank"
      >
        <img
          alt="React-logo"
          class="logoImg"
          src="React-logo.png"
        />
      </a>
      <a
        href="https://nodejs.org/en/"
        target="blank"
      >
        <img
          alt="Nodejs-logo"
          class="logoImg"
          src="Nodejs-logo.png"
        />
      </a>
      <a
        href="https://www.mysql.com/"
        target="blank"
      >
        <img
          alt="MySQL-logo"
          class="logoImg"
          src="MySQL-logo.png"
        />
      </a>
    </div>
  </div>
  <div
    class="projectContainer"
    data-testid="projectContainer"
  >
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="bringAndShareFront"
        >
          <h3>
            Bring and Share
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-users fa-w-20 iconContainer"
            data-icon="users"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="bringAndShareBack"
        >
          <img
            alt="Bring and Share"
            src="BandS.png"
          />
          <p
            class="projectPhar"
          >
            This app has been a group project where I have been fully responsible to set up the API, database, and write tests. The App is allow you to create and event and add on participants with their diatry requriments. I have created the component for adding participants, and the participant list with all their details.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/Bring-and-Share"
              target="blank"
            >
              Front-End
            </a>
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/bring-and-share-api"
              target="blank"
            >
              Back-End
            </a>
            <a
              class="aCardLink"
              href="https://bring-and-share-final-project.vercel.app/"
              target="blank"
            >
              Website
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="surrealFront"
        >
          <h3>
            Surreal Estate
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-fort-awesome fa-w-16 iconContainer"
            data-icon="fort-awesome"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="surrealBack"
        >
          <img
            alt="Surreal Estate"
            src="SurrealE.png"
          />
          <p
            class="projectPhar"
          >
            It is a property search engine just like Zoopla or Rightmove. There are properties, favourites and add property pages.The user is able to log in with their Facebook account. Once logged in, you are able to save properties to your favourites or remove them.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/surreal-estate"
              target="blank"
            >
              View Source
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="weatherFront"
        >
          <h3>
            Weather App
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-sun fa-w-20 iconContainer"
            data-icon="cloud-sun"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M575.2 325.7c.2-1.9.8-3.7.8-5.6 0-35.3-28.7-64-64-64-12.6 0-24.2 3.8-34.1 10-17.6-38.8-56.5-66-101.9-66-61.8 0-112 50.1-112 112 0 3 .7 5.8.9 8.7-49.6 3.7-88.9 44.7-88.9 95.3 0 53 43 96 96 96h272c53 0 96-43 96-96 0-42.1-27.2-77.4-64.8-90.4zm-430.4-22.6c-43.7-43.7-43.7-114.7 0-158.3 43.7-43.7 114.7-43.7 158.4 0 9.7 9.7 16.9 20.9 22.3 32.7 9.8-3.7 20.1-6 30.7-7.5L386 81.1c4-11.9-7.3-23.1-19.2-19.2L279 91.2 237.5 8.4C232-2.8 216-2.8 210.4 8.4L169 91.2 81.1 61.9C69.3 58 58 69.3 61.9 81.1l29.3 87.8-82.8 41.5c-11.2 5.6-11.2 21.5 0 27.1l82.8 41.4-29.3 87.8c-4 11.9 7.3 23.1 19.2 19.2l76.1-25.3c6.1-12.4 14-23.7 23.6-33.5-13.1-5.4-25.4-13.4-36-24zm-4.8-79.2c0 40.8 29.3 74.8 67.9 82.3 8-4.7 16.3-8.8 25.2-11.7 5.4-44.3 31-82.5 67.4-105C287.3 160.4 258 140 224 140c-46.3 0-84 37.6-84 83.9z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="weatherBack"
        >
          <img
            alt="Weather App Project"
            src="weather.png"
          />
          <p
            class="projectPhar"
          >
            By using external data the app allows the user to search by cities. Once you type in the requested city it displays 5 days of weather forecast. By clicking on More Details, it gives you a more detailed description of the weather of the chosen day.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/weather-app"
              target="blank"
            >
              View Source
            </a>
            <a
              class="aCardLink"
              href="https://weather-app-ochre-two.vercel.app/"
              target="blank"
            >
              Website
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="nasaFront"
        >
          <h3>
            Nasa's Picture Search
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-rocket fa-w-16 iconContainer"
            data-icon="rocket"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505.12019,19.09375c-1.18945-5.53125-6.65819-11-12.207-12.1875C460.716,0,435.507,0,410.40747,0,307.17523,0,245.26909,55.20312,199.05238,128H94.83772c-16.34763.01562-35.55658,11.875-42.88664,26.48438L2.51562,253.29688A28.4,28.4,0,0,0,0,264a24.00867,24.00867,0,0,0,24.00582,24H127.81618l-22.47457,22.46875c-11.36521,11.36133-12.99607,32.25781,0,45.25L156.24582,406.625c11.15623,11.1875,32.15619,13.15625,45.27726,0l22.47457-22.46875V488a24.00867,24.00867,0,0,0,24.00581,24,28.55934,28.55934,0,0,0,10.707-2.51562l98.72834-49.39063c14.62888-7.29687,26.50776-26.5,26.50776-42.85937V312.79688c72.59753-46.3125,128.03493-108.40626,128.03493-211.09376C512.07526,76.5,512.07526,51.29688,505.12019,19.09375ZM384.04033,168A40,40,0,1,1,424.05,128,40.02322,40.02322,0,0,1,384.04033,168Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="nasaBack"
        >
          <img
            alt="Nasa Search Project"
            src="nasa.png"
          />
          <p
            class="projectPhar"
          >
            This is my first React web application I have built. The idea is to search images based on a query realiting to space using the NASA API.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/Nasa-picture-searchengine"
              target="blank"
            >
              View Source
            </a>
            <a
              class="aCardLink"
              href="https://tech-test.vercel.app/"
              target="blank"
            >
              Website
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="cruiseFront"
        >
          <h3>
            Cruise Ship
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-anchor fa-w-18 iconContainer"
            data-icon="anchor"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.971 352h32.394C67.172 454.735 181.944 512 288 512c106.229 0 220.853-57.38 242.635-160h32.394c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0l-67.029 67.029c-7.56 7.56-2.206 20.485 8.485 20.485h35.146c-20.29 54.317-84.963 86.588-144.117 94.015V256h52c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-52v-5.47c37.281-13.178 63.995-48.725 64-90.518C384.005 43.772 341.605.738 289.37.01 235.723-.739 192 42.525 192 96c0 41.798 26.716 77.35 64 90.53V192h-52c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v190.015c-58.936-7.399-123.82-39.679-144.117-94.015h35.146c10.691 0 16.045-12.926 8.485-20.485l-67.029-67.029c-4.686-4.686-12.284-4.686-16.971 0L4.485 331.515C-3.074 339.074 2.28 352 12.971 352zM288 64c17.645 0 32 14.355 32 32s-14.355 32-32 32-32-14.355-32-32 14.355-32 32-32z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="cruiseBack"
        >
          <img
            alt="Cruise Ship Project"
            src="cruise.png"
          />
          <p
            class="projectPhar"
          >
            At this project I have built a cruise ship which is able to sail. There is a board also included which tells you the current and the next port. You are also able to add ports. It has been designed to look really oldschool. Because that is what I love the most.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/cruise-ships"
              target="blank"
            >
              View Source
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="cardInner"
      >
        <div
          class="cardFront"
          data-testid="petFront"
        >
          <h3>
            Virtual Pet
          </h3>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-dog fa-w-18 iconContainer"
            data-icon="dog"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M298.06,224,448,277.55V496a16,16,0,0,1-16,16H368a16,16,0,0,1-16-16V384H192V496a16,16,0,0,1-16,16H112a16,16,0,0,1-16-16V282.09C58.84,268.84,32,233.66,32,192a32,32,0,0,1,64,0,32.06,32.06,0,0,0,32,32ZM544,112v32a64,64,0,0,1-64,64H448v35.58L320,197.87V48c0-14.25,17.22-21.39,27.31-11.31L374.59,64h53.63c10.91,0,23.75,7.92,28.62,17.69L464,96h64A16,16,0,0,1,544,112Zm-112,0a16,16,0,1,0-16,16A16,16,0,0,0,432,112Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="cardBack"
          data-testid="petBack"
        >
          <img
            alt="Virtual Pet Project"
            src="pet.png"
          />
          <p
            class="projectPhar"
          >
            I have built a Virtual Pet with JavaScript. Same principles like the good old game Tamagotchi. The Pet gets older and as it gets older, it gets hungrier and more and more unfit. You are able to take it for a walk to get back its fitness and of course feed it. However, nature is cruel and even if you take care of it on the best way the pet will D-I-E when it reaches 30 years.
          </p>
          <div
            class="linksCard"
          >
            <a
              class="aCardLink"
              href="https://github.com/szabinakov/Virtual-Pet"
              target="blank"
            >
              View Source
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
